_episodes/05b-advanced-containers.md

```diff
@@ -268,27 +266,76 @@ $ docker run alpine-sum:v1
 ~~~
 {: .language-bash}
 
-- Make the `sum.py` script run automatically with arguments from the command line:
+You notice that you can run the container without arguments just fine,
+resulting in `sum = 0`, but this is boring. Supplying argumens however
+doesn't work:
+~~~
+docker run  alpine-sum:v1 10 11 12
+~~~
+{: .language-bash}
+results in
+~~~
+docker: Error response from daemon: OCI runtime create failed:
+container_linux.go:349: starting container process caused "exec:
+\"10\": executable file not found in $PATH": unknown.
+~~~
+{: .output}
+
+This is because the arguments `10 11 12` are interpreted as a
+*command* that replaces the default command given by `CMD
+["python3", "/home/sum.py"]` in the image.
+
+To achieve the goal of having a command that *always* runs when the
+container is run *and* can be amended with arguments given on the
+command-line, use the keyword `ENTRYPOINT` in the `Dockerfile`.
 
 ~~~
 FROM alpine
 
 COPY sum.py /home
-RUN apk add --update python py-pip python-dev
+RUN apk add --update python3 py3-pip python3-dev
 
 # Run the sum.py script as the default command and
 # allow people to enter arguments for it
-ENTRYPOINT ["python", "/home/sum.py"]
+ENTRYPOINT ["python3", "/home/sum.py"]
+
+# Give default arguments, in case none are supplied on
+# the command-line
+CMD ["10", "11"]
 ~~~
 
 Build and test it:
 ~~~
 $ docker build -t alpine-sum:v2 .
-$ docker run alpine-sum:v2 1 2 3 4
+# Most of the time you are interested in sum of 10 and 11:
+$ docker run alpine-sum:v2
+# Sometimes you have more challenging calculations to do:
+$ docker run alpine-sum:v2 12 13 14
 ~~~
 {: .language-bash}
 
-- Add the `sum.py` script to the `PATH` so you can run it directly:
+> ## Overriding the ENTRYPOINT
```

Comment:
this is great! I think this is the perfect place for this for now.

Reply:
Thanks!